src/main/java/com/minecolonies/coremod/commands/colonycommands/DeleteColonyCommand.java

```diff
@@ -87,45 +135,56 @@ public void execute(@NotNull final MinecraftServer server, @NotNull final IComma
             {
-                canDelete = Boolean.parseBoolean(args[2]);
+                confirmDelete = Boolean.parseBoolean(args[2]);
             }
         }
 
-        if(!canDelete)
+        final Colony colony = ColonyManager.getColony(colonyId);
+        if (colony == null)
         {
-            final ITextComponent deleteButton = new TextComponentString("[DELETE]")
-                    .setStyle(new Style().setBold(true).setColor(TextFormatting.GOLD).setClickEvent(
-                            new ClickEvent(ClickEvent.Action.RUN_COMMAND, "/mc colony delete " + colonyId + " " + canDestroy + " true")
-                    ));
-            sender.sendMessage(new TextComponentString("Click [DELETE] to confirm the deletion of colony: " + colonyId));
-            sender.sendMessage(deleteButton);
+            final String noColonyFoundMessage = String.format(NO_COLONY_FOUND_FOR_ID_MESSAGE, colonyId);
+            sender.sendMessage(new TextComponentString(noColonyFoundMessage));
             return;
         }
 
+        executeShared(server, sender, colony, canDestroy, confirmDelete);
+    }
 
-        final Colony colony = ColonyManager.getColony(colonyId);
-        if (colony == null)
+    private void executeShared(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, final Colony colony, final boolean canDestroy,
+            final boolean confirmDelete) throws CommandException
+    {
+
+        if (!confirmDelete)
         {
-            final String noColonyFoundMessage = String.format(NO_COLONY_FOUND_MESSAGE_ID, colonyId);
-            sender.sendMessage(new TextComponentString(noColonyFoundMessage));
+            final ITextComponent deleteButton = new TextComponentString("[DELETE]")
+                    .setStyle(new Style().setBold(true).setColor(TextFormatting.GOLD).setClickEvent(
+                            new ClickEvent(ClickEvent.Action.RUN_COMMAND,
+                                    String.format(DELETE_COLONY_CONFIRM_DELETE_COMMAND_SUGGESTED,
+                                            colony.getID(), (canDestroy ? "true" : "false"))
+                    )));
+            sender.sendMessage(new TextComponentString("Click [DELETE] to confirm the deletion of colony: " + colony.getID()));
+            sender.sendMessage(deleteButton);
             return;
         }
 
+
         final Entity senderEntity = sender.getCommandSenderEntity();
 
         if (senderEntity instanceof EntityPlayer)
         {
             final EntityPlayer player = (EntityPlayer) sender;
-            if (!canPlayerUseCommand(player, DELETECOLONY, colonyId))
+            if (!canPlayerUseCommand(player, DELETECOLONY, colony.getID()))
             {
                 senderEntity.sendMessage(new TextComponentString(NOT_PERMITTED));
                 return;
             }
         }
         final boolean shouldDestroy = canDestroy;
+        // TODO: pass in sender and notify when the delete task finishes.
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)